fe/fe-core/src/main/java/org/apache/doris/task/PushTask.java

```diff
@@ -127,33 +130,36 @@ public TPushReq toThrift() {
                 break;
             case DELETE:
                 List<TCondition> tConditions = new ArrayList<TCondition>();
+                Map<String, TColumn> colNameToColDesc = columnsDesc.stream()
+                        .collect(Collectors.toMap(TColumn::getColumnName, Function.identity()));
                 for (Predicate condition : conditions) {
                     TCondition tCondition = new TCondition();
                     ArrayList<String> conditionValues = new ArrayList<String>();
+                    SlotRef slotRef = (SlotRef) condition.getChild(0);
+                    String columnName = slotRef.getColumnName();
+                    tCondition.setColumnName(columnName);
+                    int uniqueId = colNameToColDesc.get(columnName).getColUniqueId();
+                    if (uniqueId >= 0) {
+                        tCondition.setColumnUniqueId(uniqueId);
+                    }
```

Comment:
please also modify code in BE and check the column unique id exists in tablet schema and column name is equal to the column name in delete predicate.
The code maybe in push handler.cpp